### PR TITLE
fix(navigation): revert css style for state-locked nav

### PR DESCRIPTION
**Pull request by FelixBlaisThon**
### Proposed Changes

reverting this https://github.com/coveo/plasma/pull/2669. instead I used the disabled props from the SideNavigationItem and it works like a charm

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Review by aiheon — APPROVED** [review]

File: packages/style/scss/components/navigation.scss [hunk 1]
```diff
@@ -148,7 +147,10 @@
 
                             &.state-locked {
                                 color: $navigation-locked-color;
-                                pointer-events: none;
+                                cursor: default;
+                                &:hover {
+                                    background: none;
+                                }
```
Comment (GermainBergeron) [inline packages/style/scss/components/navigation.scss:153]: That's for the SVG?

**Review by GermainBergeron — COMMENTED** [review]
Do you have a code snippet we can put in the sandbox to test it out?

**Review by mikedidomizio — APPROVED** [review]

**Comment by mikedidomizio** [comment]
CSS wise it is fine.  I've approved with the resolution of Germain's question.

**Comment by FelixBlaisThon** [comment]
here is th code snippet to copy paste in sidebar navigation (first example) in plasma 
```import {SideNavigation, SideNavigationItem, SideNavigationMenuSection, Tooltip, Svg} from '@coveord/plasma-react';
import {FunctionComponent} from 'react';

const Item: FunctionComponent<{isLocked: boolean; url: string; currentPath: string; disabled?: boolean}> = ({isLocked, url, currentPath, disabled, children}) => {
    return (
        <SideNavigationItem isActive={currentPath.startsWith(url)} disabled={disabled}>
            <a href={url} className={!!isLocked ? 'state-locked' : ''} style={{height: '35px'}}>
                <span className="navigation-menu-section-item-link-name nowrap">{children}</span>
                {isLocked ? (
                    <Tooltip title={'Label'}>
                        <Svg
                            svgName="lock"
                            svgClass="icon mod-16 no-link"
                            className="navigation-menu-section-item-link-icon"
                        />
                    </Tooltip>
                ) : null}
            </a>
        </SideNavigationItem>
    )
}

export default () => (
    <SideNavigation>
        <SideNavigationMenuSection title="Regular Section" svgName={'coveoIcon16px'}>
            <Item url="#coveo.com" currentPath="hello-world" isLocked disabled>
                Hello
            </Item>
            <Item url="#coveo.com" currentPath="hello-world" isLocked={false}>
                Hello
            </Item>
        </SideNavigationMenuSection>
    </SideNavigation>
);```

File: packages/style/scss/components/navigation.scss [hunk 2]
```diff
@@ -148,7 +147,10 @@
 
                             &.state-locked {
                                 color: $navigation-locked-color;
-                                pointer-events: none;
+                                cursor: default;
+                                &:hover {
+                                    background: none;
```
[the remark this answers is not in the document]
Comment (FelixBlaisThon) [reply packages/style/scss/components/navigation.scss:152]: You're right, disabled already covers it. I'll remove it

**Pull request merged**